src/string.cpp

```diff
@@ -368,6 +381,70 @@ bool StrEndsWith(const std::string_view str, const std::string_view suffix)
 	return str.compare(str.size() - suffix_len, suffix_len, suffix, 0, suffix_len) == 0;
 }
 
+/**
+ * Check whether the given string ends with the given suffix, ignoring case.
+ * @param str    The string to look at.
+ * @param suffix The suffix to look for.
+ * @return True iff the end of the string is the same as the suffix, ignoring case.
+ */
+bool StrEndsWithIgnoreCase(std::string_view str, const std::string_view suffix)
+{
+	if (str.size() < suffix.size()) return false;
+	return StrEqualsIgnoreCase(str.substr(str.size() - suffix.size()), suffix);
+}
+
+/** Case insensitive implementation of the standard character type traits. */
+struct ci_char_traits : public std::char_traits<char> {
```

Comment:
why not just write it out? 
```suggestion
struct case_insensitive_char_traits : public std::char_traits<char> {
```